### DISCUSSION TITLE
aws-s3-deployment: Creates a custom lambda resource in Python 3.6 which is no longer supported

### DISCUSSION THREAD

**Discussion by shwetajoshi601** [aws/aws-cdk#20491] [Q&A]
Hello,

We are using version 1.112.0 for all our aws cdk modules. The aws-s3-deployment modules is used for creating a serverless deployment bucket.

The code used for the custom resource is using the Python 3.6 runtime of AWS Lambda, this is something internal to the module.

AWS Lambda is no longer going support updates to functions using Python 3.6, how do we make this module create a function in versions higher than 3.6?

Refer to the development section of this version.
[v1.112.0](https://www.npmjs.com/package/@aws-cdk/aws-s3-deployment/v/1.112.0)

**Comment by cgarvis** [comment]
Hi @shwetajoshi601 we are aware of the deprecation of Python 3.6 by Lambda.  We have a #19988 in the works.  Once released, you should be able to upgrade CDK and redeploy.

**Reply by shwetajoshi601** [reply]
We will be waiting for the update. Thank you so much!
